### PR TITLE
fix(auth): tries to reauthenticate once

### PR DESCRIPTION
**Pull request by GustaafL**

### CODE REVIEW

File: src/flexmeasures_client/response_handling.py [hunk 1]
```diff
@@ -26,10 +26,12 @@ async def check_response(self: FlexMeasuresClient, response, polling_step: int):
     print(status)
     if status < 300:
         pass
-    elif status == 401:
+    elif status == 401 and payload.get("status") == "UNAUTHORIZED":
```
Comment (Flix6x) [inline src/flexmeasures_client/response_handling.py:29]: I was expecting this to be the fix:
```
elif  status == 401 and self.reauth_once:
```
Why the check for UNAUTHORIZED instead?

Comment (GustaafL) [reply src/flexmeasures_client/response_handling.py:29]: That solution does not work. `self.reauth` is set to true in `request`. So once it tries to execute the request again it will try to reauthenticate. 

UNAUTHORIZED is the status response if the wrong `self.access_token` is used for a request, triggering a reauth which can not fail in the same way because `get_access_token()` doesn't use the access_token.

Comment (Flix6x) [reply src/flexmeasures_client/response_handling.py:29]: It seems like a rather implicit way of solving the issue. The goal is to prevent get_access_token from triggering itself in a loop, right?

An alternative could be to only attempt to reauthenticate if the auth header is used. This could be done by setting reauth_once based on include_auth. In fact, we may just as well use include_auth instead of reauth_once to decide whether to attempt to reauthenticate. And that would also mean we can get rid of setting up reauth_once as a class parameter.

Comment (Flix6x) [reply src/flexmeasures_client/response_handling.py]: I still found the UNAUTHORIZED check too implicit, which worried me that future developers will find it hard to know what this is for. I also found the `self.headers` approach risky with possible the wrong headers used when different methods are called simultaneously, and multiple polling loops ongoing. I found the trick was to move the `get_headers` into the polling loop and actually connecting the incremented polling step, which had been disconnected in `request_once`.

The only thing that doesn't look too elegant now is the passing back and forth of the `polling_step` and the `reauth_once` parameters. Perhaps at some point we can create a local request_context variable or something, where we can store this info, and update them inplace. But that's for another time.

To summarize, I feel this new approach is not too elegant, but more explicit, and therefore has my preference.

File: src/flexmeasures_client/client.py [hunk 2]
```diff
@@ -101,20 +101,26 @@ async def request(
         self.start_session()
 
         polling_step = 0  # reset this counter once when starting polling
-        self.reauth_once = True
+        # we allow retrying once if we include authentication headers
+        reauth_once = self.reauth_once if include_auth else False
```
Comment (GustaafL) [inline src/flexmeasures_client/client.py:105]: Do we want users to be able to turn of the reauthentication? otherwise we can just use `True` instead of self.reauth_once here

**Review by GustaafL — COMMENTED** [review]
Looks a lot cleaner! Thanks

File: src/flexmeasures_client/client.py (hunk 2 above)
Comment (Flix6x) [reply src/flexmeasures_client/client.py:105]: I can't think of a use case either. I'll remove it from the initialization settings.

**Review by Flix6x — APPROVED** [review]
